### DISCUSSION TITLE
Couple Ideas - Days of week & Test Notifications

### DISCUSSION THREAD

**Discussion by steffanvigano** [juftin/camply#226] [Ideas]
Would love to see the following in the future:

Day Of Week Arrival:
-I see the weekends flag, but I work in the resort industry and weekends are for other people.  Preferred would be day of week arrival.  IE, I might want a site for 2 nights, between certain dates, and I can arrive either on a Tues or a Wed. 

Test Notification Flag:
-With all campground selling out almost instantly, and all my trial runs coming back empty, it would be nice to be able to test out notifications rather than waiting for an eventual hit. 

Date Windows:
-Rather than search across 4 months worth of dates, If I'm really only available to camp a couple specific weeks, it would be nice to only search those specific date ranges.   IE. allow multiple, non overlapping, start and end dates.  Currently in a YAML file, only the last defined dates are used.   (I know I could just set up multiple YAML files and run different searches, so maybe this isn't needed)

Thanks!

**Comment by juftin** [comment]
@steffanvigano really great ideas! Here's a small feature that gets you part of the way there - just released on `camply 0.15.0`: Multiple Search Windows!

```
camply campsites \
    --rec-area 2725 \
    --start-date 2023-07-12 \
    --end-date 2023-07-13 \
    --start-date 2023-07-19 \
    --end-date 2023-07-20 \
    --start-date 2023-07-26 \
    --end-date 2023-07-27
```

https://juftin.com/camply/command_line_usage/#searching-across-multiple-time-windows

**Comment by steffanvigano** [comment]
Thanks for the quick turnaround.  FYI that in my testing at least, it looks like it works from command line, but not while reading a yaml file.
-S

**Reply by juftin** [reply]
Check out this YAML file, I've got it working in the test suite https://github.com/juftin/camply/blob/7864c3856dc1a5a3e7e726243e822bf947579f91/tests/yaml/double_window_campsite_search.yaml

**Reply by steffanvigano** [reply]
Ahh...  I figured that following the command line syntax was logical so I had start/end/start/end/start/end one after another in order inside the yaml file.  Working now.  Thanks.

**Comment by juftin** [comment]
Alright, @steffanvigano I can officially call these feature requests complete. These were fun and easy to code up. 

The following features were implemented in camply `0.15.0` and now `0.16.0`:
- [✨ test-notifications](https://juftin.com/camply/command_line_usage/#test-notifications)
- [✨ Search By Weekday](https://juftin.com/camply/command_line_usage/#searching-for-specific-weekdays)
- [✨ Multiple Search Windows](https://juftin.com/camply/command_line_usage/#searching-across-multiple-time-windows)

I'll try to do a better job of documenting YAML usage soon - for now there's somewhat helpful inline comments located on the [YAML section of the docs](https://juftin.com/camply/command_line_usage/#using-a-yaml-configuration-file-to-search-for-campsites)

**Reply by steffanvigano** [reply]
Amazing response time.  All working as expected, and especially like the sample link in the test email. 

Thanks and cheers to a great upcoming camping season!  Ping me if you come to Tahoe and I'll buy you a beer.

**Comment by steffanvigano** [comment]
Hi again, 
After using these for a while, I realize that it actually might work better if one could EXCLUDE either days of the week, or specific dates.   

Example:
    --start-date 2023-09-10 \
    --end-date 2023-11-11 \
    --exclude-date 2023-09-13 \
    --exclude-date 2023-10-10 \
    --exclude-day Tuesday \

YAML Example:
    start_date:
        - 2023-09-10 # YYYY-MM-DD
    end_date:
        - 2023-11-11 # YYYY-MM-DD
    exclude_date:
        - 2023-10-10 # YYYY-MM-DD
    exclude_day:
        - Tuesday

Obviously a nice to have.  Thanks again for such a great tool. 
-S

**Reply by juftin** [reply]
Oh, I'll keep this in mind, thank you! In the meantime you can achieve this with [Search By Weekday](https://juftin.com/camply/command_line_usage/#searching-for-specific-weekdays) and [Multiple Search Windows](https://juftin.com/camply/command_line_usage/#searching-across-multiple-time-windows) - in a slightly more verbose way 😄

**Comment by steffanvigano** [comment]
I've tried the -include dates, but I got some odd results just now.  I got notified for a site available on a Tuesday even though I have this in my YAML file:

day: Thursday
day: Friday
day: Saturday
day: Sunday
day: Monday

Perhaps I'm getting too complex?  Or they need to be in order?  

Full YAML, Log & Notification details are here:   https://pastebin.com/pz9XCbuN

Thanks
